megalinter/MegaLinter.py

```diff
@@ -428,15 +428,9 @@ def list_files_all(self):
         ]
         if logging.getLogger().isEnabledFor(logging.DEBUG):
             logging.debug("Root dir content:\n" + "\n- ".join(all_files))
-        excluded_directories = utils.list_excluded_directories()
-        for (dirpath, _dirnames, filenames) in os.walk(self.workspace):
-            exclude = False
-            for dir1 in dirpath.split(os.path.sep):
-                if dir1 in excluded_directories:
-                    exclude = True
-                    break
-            if exclude is True:
-                continue
+        excluded_directories = set(config.get("EXCLUDED_DIRECTORIES", utils.list_excluded_directories()))
```

Comment:
use config.get_list , so it can manage variables from ENV and from .mega-linter.yml